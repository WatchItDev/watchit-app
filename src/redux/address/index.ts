import { createSlice, PayloadAction } from '@reduxjs/toolkit';

interface AddressState {
  address: string;
  profileId: string;
  showRainbow: boolean;
}

const initialState: AddressState = {
  address: '',
  profileId: '',
  showRainbow: false,
};

const addressSlice = createSlice({
  name: 'address',
  initialState,
  reducers: {
    storeAddress(state, action: PayloadAction<{ address: string; profileId: string }>) {
      state.address = action.payload.address;
      state.profileId = action.payload.profileId;
    },

    toggleRainbow(state) {
      state.showRainbow = !state.showRainbow;
    },

  },
});

export const { storeAddress, toggleRainbow } = addressSlice.actions;
export default addressSlice.reducer;
