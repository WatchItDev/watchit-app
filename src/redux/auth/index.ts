import {createSlice, PayloadAction} from "@reduxjs/toolkit";

export interface AuthReducerState {
  session: any;
  isSessionLoading: boolean;
  isLoginModalOpen: boolean;
  balance: number;
  currentStep: number;
  isUpdatingMetadata: boolean;
  email: string;
}

const initialState: AuthReducerState = {
  session: null,
  isSessionLoading: false,
  isLoginModalOpen: false,
  balance: 0,
  currentStep: 0,
  isUpdatingMetadata: false,
  email: "",
};

const authSlice = createSlice({
  name: "auth",
  initialState,
  reducers: {
    openLoginModal: (state) => {
      state.isLoginModalOpen = true;
    },
    closeLoginModal: (state) => {
      state.isLoginModalOpen = false;
    },
    setAuthLoading: (state, action: PayloadAction<Pick<AuthReducerState, "isSessionLoading">>) => {
      state.isSessionLoading = action.payload.isSessionLoading;
    },
    setBalance: (state, action: PayloadAction<Pick<AuthReducerState, "balance">>) => {
      state.balance = action.payload.balance;
    },
    setProfileCreationStep: (state, action: PayloadAction<{step: number}>) => {
      state.currentStep = action.payload.step;
    },
    resetCurrentStep: (state) => {
      state.currentStep = 0;
    },
    setSession: (state, action: PayloadAction<{session: any}>) => {
      state.session = action.payload.session;
    },
    setEmail: (state, action: PayloadAction<string>) => {
      state.email = action.payload;
    },
    updateProfileData: (
      state,
      action: PayloadAction<{
        name: string;
        bio: string;
        profileImage?: string;
        backgroundImage?: string;
      }>,
    ) => {
      if (state.session && state.session.profile) {
        state.session.profile.metadata = {
          ...state.session.profile.metadata,
          displayName: action.payload.name,
          bio: action.payload.bio,
          picture: {
            optimized: {
              uri: action.payload.profileImage ? action.payload.profileImage : undefined,
            },
          },
          coverPicture: {
            optimized: {
              uri: action.payload.backgroundImage ? action.payload.backgroundImage : undefined,
            },
          },
        };
      }
    },
    setIsUpdatingMetadata: (state, action: PayloadAction<boolean>) => {
      state.isUpdatingMetadata = action.payload;
    },
  },
});

export const {
  openLoginModal,
  closeLoginModal,
  setAuthLoading,
  setBalance,
  setProfileCreationStep,
  resetCurrentStep,
  setSession,
  updateProfileData,
  setIsUpdatingMetadata,
  setEmail,
} = authSlice.actions;

export default authSlice.reducer;
