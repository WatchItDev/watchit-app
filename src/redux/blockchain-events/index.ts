import { createSlice, PayloadAction } from '@reduxjs/toolkit'
import { Log } from 'viem'

interface BlockchainEventsState {
  events: Log[];
}

const initialState: BlockchainEventsState = {
  events: [],
}

const blockchainEventsSlice = createSlice({
  name: 'blockchainEvents',
  initialState,
  reducers: {
    addBlockchainEvent(state, action: PayloadAction<Log>) {
      state.events.push(action.payload)
    },
    setBlockchainEvents(state, action: PayloadAction<Log[]>) {
      state.events = action.payload
    },
    clearBlockchainEvents(state) {
      state.events = []
    },
  },
})

export const { addBlockchainEvent, setBlockchainEvents, clearBlockchainEvents } =
  blockchainEventsSlice.actions

export default blockchainEventsSlice.reducer
