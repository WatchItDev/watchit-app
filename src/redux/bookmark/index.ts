import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { Post } from '@src/graphql/generated/graphql.ts';

export interface BookmarkReducerState {
  bookmarkPublications: Post[];
  hiddenBookmarks: Post[];
}

const initialState: BookmarkReducerState = {
  bookmarkPublications: [],
  hiddenBookmarks: [],
};

const bookmarkSlice = createSlice({
  name: 'bookmark',
  initialState,
  reducers: {
    // Add a post to bookmarks and remove it from hidden if it is
    addBookmark: (state, action: PayloadAction<Post>) => {
      const isAlreadyBookmarked = state.bookmarkPublications.some(
        (publication) => publication.id === action.payload.id
      );

      if (!isAlreadyBookmarked) {
        state.bookmarkPublications.push(action.payload);
        // Remove from hiddenBookmarks if exists
        state.hiddenBookmarks = state.hiddenBookmarks.filter(
          (publication) => publication.id !== action.payload.id
        );
      }
    },

    // Move a post to hiddenBookmarks instead of deleting it completely
    removeBookmark: (state, action: PayloadAction<string>) => {
      const publicationToHide = state.bookmarkPublications.find(
        (publication) => publication.id === action.payload
      );

      if (publicationToHide) {
        state.hiddenBookmarks.push(publicationToHide);
        state.bookmarkPublications = state.bookmarkPublications.filter(
          (publication) => publication.id !== action.payload
        );
      }
    },

    // Remove all bookmarks
    clearBookmarks: (state) => {
      state.bookmarkPublications = [];
    },
  },
});

export const { addBookmark, removeBookmark, clearBookmarks } = bookmarkSlice.actions;

export default bookmarkSlice.reducer;
