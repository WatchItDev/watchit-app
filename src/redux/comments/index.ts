import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { AnyPublication } from '@lens-protocol/api-bindings';

type PendingComment = AnyPublication & { uri: string };

export type CommentsReducerState = {
  refetchTriggerByPublication: {
    [publicationId: string]: number;
  };
  hiddenComments: AnyPublication[];
  counterLikes: { [publicationId: string]: number };
  comments: { [publicationId: string]: AnyPublication[] };
  pendingComments: { [publicationId: string]: PendingComment[] };
};

const initialState: CommentsReducerState = {
  refetchTriggerByPublication: {},
  hiddenComments: [],
  counterLikes: {},
  comments: {},
  pendingComments: {},
};

const commentsSlice = createSlice({
  name: 'comments',
  initialState,
  reducers: {
    refetchCommentsByPublication: (state, action: PayloadAction<string>) => {
      const publicationId = action.payload;
      if (!state.refetchTriggerByPublication[publicationId]) {
        state.refetchTriggerByPublication[publicationId] = 1;
      } else {
        state.refetchTriggerByPublication[publicationId] += 1;
      }
    },
    hiddeComment: (state, action: PayloadAction<AnyPublication>) => {
      state.hiddenComments.push(action.payload);
    },
    setCounterLikes: (state, action: PayloadAction<{ publicationId: string; likes: number }>) => {
      state.counterLikes[action.payload.publicationId] = action.payload.likes;
    },
    incrementCounterLikes: (state, action: PayloadAction<string>) => {
      const publicationId = action.payload;
      if (state.counterLikes[publicationId] !== undefined) {
        state.counterLikes[publicationId] += 1;
      }
    },
    decrementCounterLikes: (state, action: PayloadAction<string>) => {
      const publicationId = action.payload;
      if (state.counterLikes[publicationId] !== undefined) {
        state.counterLikes[publicationId] -= 1;
      }
    },
    addPendingComment: (
      state,
      action: PayloadAction<{ publicationId: string; comment: PendingComment }>
    ) => {
      const { publicationId, comment } = action.payload;
      if (!state.pendingComments[publicationId]) {
        state.pendingComments[publicationId] = [];
      }
      // Prepend new comment to the beginning of the list
      state.pendingComments[publicationId] = [comment, ...state.pendingComments[publicationId]];
    },
    removePendingComment: (
      state,
      action: PayloadAction<{ publicationId: string; commentId: string }>
    ) => {
      const { publicationId, commentId } = action.payload;

      // Delete the comment from the pending list
      state.pendingComments[publicationId] = state.pendingComments[publicationId].filter(
        (comment) => comment.id !== commentId
      );
    },
  },
});

export const {
  hiddeComment,
  refetchCommentsByPublication,
  setCounterLikes,
  incrementCounterLikes,
  decrementCounterLikes,
  addPendingComment,
  removePendingComment,
} = commentsSlice.actions;

export default commentsSlice.reducer;
