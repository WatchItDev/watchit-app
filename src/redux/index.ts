export { minibarInitialState } from "@redux/minibar";
export { drawerInitialState } from "@redux/drawer";
export { authInitialState } from "@redux/auth";
export { bookmarkInitialState } from "@redux/bookmark";
export { commentsInitialState } from "@redux/comments";
export { notificationsInitialState } from "@redux/notifications";
export { followersInitialState } from "@redux/followers";
export { addressInitialState } from "@redux/address";
export { blockchainEventsInitialState } from "@redux/blockchain-events";
export { transactionInitialState } from "@redux/transactions";
