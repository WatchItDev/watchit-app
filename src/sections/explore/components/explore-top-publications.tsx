import CarouselTopTitles from '@src/components/carousel/variants/carousel-top-titles.tsx';
import { useGetRecentPostsQuery } from '@src/graphql/generated/hooks.tsx';
import { ExploreTopPublicationsSkeleton } from '@src/sections/explore/components/explore-top-publications.skeleton.tsx';
import { LoadingFade } from '@src/components/LoadingFade.tsx';

// ----------------------------------------------------------------------

export const ExploreTopPublications = () => {
  const { data: allData, loading } = useGetRecentPostsQuery({
    variables: { limit: 10 },
  });

  return (
    <LoadingFade
      loading={loading}
      skeleton={<ExploreTopPublicationsSkeleton />}
    >
      <CarouselTopTitles posts={allData?.getRecentPosts ?? []} />
    </LoadingFade>
  );
};
