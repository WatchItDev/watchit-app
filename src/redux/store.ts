import { type Store } from 'redux';
import { configureStore } from '@reduxjs/toolkit';
import rootReducer from '@src/redux/reducer';

export const createStore = (initialState = {}): Store => {
  return configureStore({
    reducer: rootReducer,
    devTools: true,
    preloadedState: initialState,
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware({
        serializableCheck: false,
      }),
  });
};

export type RootState = ReturnType<typeof store.getState>;

export const store = createStore();
