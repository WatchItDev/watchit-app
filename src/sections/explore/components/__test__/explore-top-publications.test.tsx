import { render, screen, waitFor } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import { Provider } from "react-redux";
import { vi } from "vitest";
import { store } from "@src/redux/store";
import { ExploreTopPublications } from "../explore-top-publications";

vi.mock("@src/workers/backgroundTaskWorker?worker", () => {
  return {
    default: class {
      postMessage() {}
      terminate() {}
      addEventListener() {}
      removeEventListener() {}
    },
  };
});
vi.mock("@lens-protocol/react-web", () => import("../__test__/__mocks__/lens-protocol-react-web"));

const renderWithProviders = () => {
  return render(
    <Provider store={store}>
      <MemoryRouter>
        <ExploreTopPublications />
      </MemoryRouter>
    </Provider>,
  );
};

describe("Testing in the <ExploreTopPublications/> component", () => {
  it("should match snapshot", () => {
    const { container } = renderWithProviders();
    expect(container).toMatchSnapshot();

    screen.debug(container);
  });

  // it("should render the component with content", () => {
  //   renderWithProviders();
  //   expect(screen.getByText("TÃ­tulo de prueba")).toBeInTheDocument();
  //   expect(screen.getByText("Contenido de prueba para explorar publicaciones")).toBeInTheDocument();
  // });
});
