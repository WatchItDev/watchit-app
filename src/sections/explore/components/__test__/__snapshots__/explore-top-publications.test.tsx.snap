// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing in the <ExploreTopPublications/> component > should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-14btkih"
  >
    <div
      class="MuiStack-root css-nen11g-MuiStack-root"
      data-testid="carouselArrowsContainer"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-oxvw91-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          data-icon="eva:arrow-ios-forward-fill"
          data-sx="{"transform":" scaleX(-1)"}"
          data-testid="leftButton"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="slick-slider slick-initialized"
      >
        <div
          class="slick-list"
          style="height: 0px;"
        >
          <div
            class="slick-track"
            style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
          >
            <div
              aria-hidden="false"
              class="slick-slide slick-active slick-current"
              data-index="0"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  class="MuiBox-root css-1fpx25h"
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <div
                    class="MuiStack-root css-1dw07r3-MuiStack-root"
                  >
                    <span
                      class="component-image MuiBox-root css-1k4qe2b"
                    >
                      <span
                        class="component-image-wrapper lazy-load-image-background blur"
                        style="background-image: url(/assets/placeholder.svg); background-size: 100% 100%; color: transparent; display: inline-block;"
                      >
                        <img
                          alt="Prueba de titulo"
                          class="MuiBox-root css-19t2nho"
                          src=""
                        />
                      </span>
                    </span>
                    <div
                      class="MuiBox-root css-1wf6v1o"
                    >
                      <div
                        class="MuiBox-root css-bwbmtv"
                      >
                        <div
                          class="MuiBox-root css-b8hdil"
                        >
                          <span
                            class="component-image MuiBox-root css-1jl2tef"
                            style="border-radius: 10px; width: 100%;"
                          >
                            <span
                              class="component-image-wrapper lazy-load-image-background blur"
                              style="background-image: url(/assets/placeholder.svg); background-size: 100% 100%; color: transparent; display: inline-block;"
                            >
                              <img
                                alt="Prueba de titulo"
                                class="MuiBox-root css-19t2nho"
                                src=""
                              />
                            </span>
                          </span>
                        </div>
                      </div>
                      <div
                        class="MuiBox-root css-1x26m9t"
                      >
                        <div
                          class="MuiStack-root css-1u3oz1v-MuiStack-root"
                        >
                          <div
                            class="MuiStack-root css-1fkouc0-MuiStack-root"
                          >
                            <h3
                              class="MuiTypography-root MuiTypography-h3 css-8f07vu-MuiTypography-root"
                            >
                              Prueba de titulo
                            </h3>
                            <div
                              class="MuiBox-root css-1yp4ln"
                            >
                              <button
                                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1x59mi6-MuiButtonBase-root-MuiButton-root"
                                tabindex="0"
                                type="button"
                              >
                                <span
                                  class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
                                  style="margin-right: 5px;"
                                >
                                  by
                                </span>
                                <span
                                  class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
                                  style="gap: 4px; display: flex; align-items: center; background-color: rgba(0, 0, 0, 0.7); padding: 4px 10px; border-radius: 8px;"
                                >
                                  <span
                                    class="component-image MuiBox-root css-1jl2tef"
                                    style="width: 20px; height: 20px; border-radius: 50%;"
                                  >
                                    <span
                                      class="component-image-wrapper lazy-load-image-background blur"
                                      style="background-image: url(/assets/placeholder.svg); background-size: 100% 100%; color: transparent; display: inline-block;"
                                    >
                                      <img
                                        class="MuiBox-root css-19t2nho"
                                        src="https://placehold.co/40x40"
                                      />
                                    </span>
                                  </span>
                                  Alex Talavera
                                </span>
                                <span
                                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                                />
                              </button>
                            </div>
                          </div>
                          <h6
                            class="MuiTypography-root MuiTypography-h6 css-1r23mgh-MuiTypography-root"
                          >
                            Contenido de prueba para explorar publicaciones
                          </h6>
                          <div
                            class="MuiStack-root css-kdc3n8-MuiStack-root"
                          >
                            <button
                              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-3fybcr-MuiButtonBase-root-MuiButton-root"
                              tabindex="0"
                              type="button"
                            >
                              <div
                                class="MuiBox-root css-16t8mkw"
                              >
                                <svg
                                  class="tabler-icon tabler-icon-player-play "
                                  fill="none"
                                  height="22"
                                  stroke="#000"
                                  stroke-linecap="round"
                                  stroke-linejoin="round"
                                  stroke-width="2"
                                  viewBox="0 0 24 24"
                                  width="22"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M7 4v16l13 -8z"
                                  />
                                </svg>
                              </div>
                              <span
                                class="MuiTypography-root MuiTypography-button css-o6jcpp-MuiTypography-root"
                              >
                                Watch now
                              </span>
                              <span
                                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                              />
                            </button>
                            <button
                              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1t5pcmw-MuiButtonBase-root-MuiButton-root"
                              tabindex="0"
                              type="button"
                            >
                              <svg
                                class="tabler-icon tabler-icon-bookmark "
                                fill="none"
                                height="22"
                                stroke="#FFFFFF"
                                stroke-linecap="round"
                                stroke-linejoin="round"
                                stroke-width="2"
                                viewBox="0 0 24 24"
                                width="22"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M18 7v14l-6 -4l-6 4v-14a4 4 0 0 1 4 -4h4a4 4 0 0 1 4 4z"
                                />
                              </svg>
                              <span
                                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                              />
                            </button>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-17oewfz-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          data-icon="eva:arrow-ios-forward-fill"
          data-sx="{}"
          data-testid="rightButton"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
